### ISSUE TITLE
Dataflow assertion failure in getValue(): org.checkerframework.dataflow.analysis.Analysis

### ISSUE THREAD

**Issue by dbrosoft** [typetools/checker-framework#765]
Save and run the following script:

``` sh
#!/bin/sh
cat << EOF > C.java
class C {
  Thread thread;

  void execute() {
    thread = new Thread() {
      @Override
      public void run() {}
    };
    thread.start();
  }
}
EOF
java -jar ${CHECKERFRAMEWORK}/checker/dist/checker.jar -processor nullness -AprintErrorStack C.java
```

Observed output begins with:

```
error: SourceChecker.typeProcess: unexpected Throwable (AssertionError) while processing C.java
  Compilation unit: C.java
  Exception: java.lang.AssertionError; Stack trace: org.checkerframework.dataflow.analysis.Analysis.getValue(Analysis.java:630)
```

Full stack trace: [trace.txt](https://github.com/typetools/checker-framework/files/292277/trace.txt)

**Comment by smillst** [comment]
This is an asSuper bug, which is fixed in my asSuper branch.  This issue will be resolved when #717 is.

**Comment by smillst** [comment]
A fix for the crash was commit cccfea6, but this issue should still be investigated.
